libs/domain/cart/coupon/coupon.component.ts

```diff
@@ -0,0 +1,125 @@
+import { IconTypes } from '@spryker-oryx/ui/icon';
+import { ColorType } from '@spryker-oryx/ui/link';
+import { ClearIconAppearance } from '@spryker-oryx/ui/searchbox';
+import { hydrate } from '@spryker-oryx/utilities';
+import { LitElement, TemplateResult, html } from 'lit';
+import { query, state } from 'lit/decorators.js';
+import { repeat } from 'lit/directives/repeat.js';
+import { couponStyles } from './coupon.styles';
+
+@hydrate({ event: ['window:load'] })
+export class CouponComponent extends CartComponentMixin(
+  ContentMixin(LitElement)
+) {
+  static styles = couponStyles;
+
+  @state() hasError = false;
+
+  @query('input[name=coupon]') coupon?: HTMLInputElement;
+
+  protected cartService = resolve(CartService);
+  protected notificationService = resolve(NotificationService);
+  protected localeService = resolve(LocaleService);
+
+  protected override render(): TemplateResult | void {
+    if (this.$isEmpty()) {
+      return;
+    }
+
+    return html`
+      <oryx-layout>
+        <h3>${this.i18n('coupon.have-a-coupon')}?</h3>
+
+        <section>
+          <oryx-input
+            ?hasError="${this.hasError}"
+            .errorMessage="${this.hasError
+              ? this.coupon?.value === ''
+                ? `${this.i18n('coupon.insert-a-coupon')}`
+                : `${this.i18n('coupon.cart-code-can-not-be-added')}`
+              : ''}"
+          >
+            <input
+              placeholder="${this.i18n('coupon.coupon-code')}"
+              name="coupon"
+            />
+            <oryx-icon
+              type=${IconTypes.Close}
+              class="clear-icon"
+              appearance=${ClearIconAppearance.Hover}
+              @click=${(): void => this.clear()}
+            ></oryx-icon>
+          </oryx-input>
+          <oryx-button
+            .type="${ButtonType.Outline}"
+            .color="${ColorType.Neutral}"
+            .size=${ButtonSize.Md}
+            @click=${this.onSubmit}
+          >
+            ${this.i18n('coupon.apply')}
+          </oryx-button>
+        </section>
+      </oryx-layout>
+
+      <ul>
+        ${repeat(
+          this.$coupons(),
+          (coupon) => coupon.code,
+          (coupon) => {
+            return html`
+              <li>
+                <oryx-icon type="done"></oryx-icon>
+                <span class="code">${coupon.code} </span>
+                <span class="name">
+                  ${coupon.displayName}
+                  <oryx-date
+                    .stamp=${coupon.expirationDateTime}
+                    .i18nToken=${'coupon.(valid-till-<date>)'}
+                  ></oryx-date>
+                </span>
+              </li>
+            `;
+          }
+        )}
+      </ul>
+    `;
+  }
+
+  protected clear(): void {
+    this.coupon!.value = '';
+    this.hasError = false;
+  }
+
+  protected onSubmit(): void {
+    this.hasError = false;
+
+    if (!this.coupon?.value) {
```

Comment:
We reuse `this.coupon?.value` three time here, mb it's better to move it to variable?